### ISSUE TITLE
warn on intended use of string interpolation 

### ISSUE THREAD

**Issue by jwkvam** [PyCQA/pylint#1580]
### Steps to reproduce
Try to use string interpolation but forget the `f` to indicate it's an f-string.
```
x = 1
s = '{x}' # s != '1'
```

### Current behavior
No indication of a problem.

### Expected behavior
If it doesn't generate too many false positives, I would greatly value a warning.

### pylint --version output
```
pylint 1.7.2,
astroid 1.5.3
Python 3.6.1 | packaged by conda-forge | (default, May 23 2017, 14:31:56)
[GCC 4.2.1 Compatible Apple LLVM 6.1.0 (clang-602.0.53)]
```

**Comment by rogalski** [comment]
I strongly disagree. Below there's a trivial code that which would emit a false positive.

```python
key = 12
pattern = '{key}'
s = pattern.format(key=3)
```

**Comment by jwkvam** [comment]
Makes sense. I'm willing to tolerate false positives for more checks than most perhaps. I also rarely use braces in strings without f-strings since I moved to 3.6. Consequently, I rarely use `.format` now.

**Comment by PCManticore** [comment]
What Lukasz is saying actually makes sense, this check will bring a couple of false positives to the table for little benefit. Let's close this for now.

**Comment by dset0x** [comment]
This was revived in #2507
